sdk/state/payment.go

```diff
@@ -188,9 +191,27 @@ func (c *Channel) validatePayment(ca CloseAgreement) (err error) {
 	if !ca.Details.ConfirmingSigner.Equal(c.localSigner.FromAddress()) && !ca.Details.ConfirmingSigner.Equal(c.remoteSigner) {
 		return fmt.Errorf("close agreement confirmer does not match a local or remote signer, got: %s", ca.Details.ConfirmingSigner.Address())
 	}
+	if !ca.Details.ProposingSigner.Equal(c.localSigner.FromAddress()) && !ca.Details.ProposingSigner.Equal(c.remoteSigner) {
+		return fmt.Errorf("close agreement proposer does not match a local or remote signer, got: %s", ca.Details.ConfirmingSigner.Address())
+	}
+
+	// If the close agreement payment amount is incorrect, error.
+	pa := ca.Details.PaymentAmount
+	if !c.initiatorProposedPayment(ca) {
+		pa = ca.Details.PaymentAmount * -1
+	}
+	if ca.Details.Balance-c.Balance() != pa {
```

Comment:
💡 I don't feel strongly about this, but when I think in my mind about what we're testing it's that the current balance is modified by the payment amount to reach the new balance, so I had to do a double take on the above to figure out if it was right, but the following is what my mind expected:
```suggestion
	if c.Balance() + pa == ca.Details.Balance {
```

Reply:
I had a similar thought
https://github.com/stellar/experimental-payment-channels/pull/291/commits/e5b16611ccd94ef0ea127d397990f33a245700f3